clash-ghc/src-ghc/Clash/GHC/Evaluator/Primitive.hs

```diff
@@ -4546,6 +4551,51 @@ ghcPrimStep tcm isSubj pInfo tys args mach = case primName pInfo of
                         , Left (Case splitCall n1BVTy [bvAlt])
                         ])
          _ -> Nothing
+#if MIN_VERSION_ghc(9,6,0)
+  "Data.Text.Show.$wunpackCStringAscii#"
+    | [Lit (StringLiteral addr)] <- args
+    , Text.Text (Text.ByteArray ba) _off len <- Text.pack addr
+    -> let (_,tyView -> TyConApp tupTcNm tyArgs) = splitFunForallTy ty
+           (Just tupTc) = UniqMap.lookup tupTcNm tcm
+           [tupDc] = tyConDataCons tupTc
+           ret     = mkApps (Data tupDc) (map Right tyArgs ++
+                    [Left (Literal (ByteArrayLiteral (BA.ByteArray ba)))
+                    ,Left (Literal (IntLiteral 0))
+                    ,Left (Literal (IntLiteral (toInteger len)))])
+        in reduce ret
+  "GHC.Magic.noinlineConstraint"
+    | [arg] <- args
+    -> reduce (valToTerm arg)
+  "GHC.TypeNats.withSomeSNat"
+    | Lit (NaturalLiteral n) : fun : _ <- args
+    , _:funTy:_ <- Either.rights (fst (splitFunForallTy ty))
+    , (tyView -> TyConApp snatTcNm _):_ <- Either.rights (fst (splitFunForallTy funTy))
+    , Just snatTc <- UniqMap.lookup snatTcNm tcm
+    , [snatDc] <- tyConDataCons snatTc
+    -> let nTy = LitTy (NumTy n)
+           snat = mkApps (Data snatDc) [Right nTy,Left (Literal (NaturalLiteral n))]
+           ret = mkApps (valToTerm fun) [Right nTy,Left snat]
```

Comment:
This applies `withSomeSNat` to an `SNat`, but its type signature wants an `Maybe (SNat n)`. Is this right?